doc/contrib/donate.rst

```diff
@@ -0,0 +1,44 @@
+.. _donation_policy:
+
+Donations
```

Comment:
Thanks for writing this doc.  I'm aware of the fact that I might be running too much tests on Jenkins even I stop the tests immediately if they are not useful.  For the extra cost I made I'm really sorry.

But for the future, here are some possibilities:  Remove automatic run of some tests, or delay it to the end.

- Like cpp linter.  I run it on my local machine but many times it returns different result than on CI.  The check is not critical, it would be better if we can put it into last step so that we can see results from other more important results without pushing a linter fix first.
- We may remove the automatic run of newer versions of cuda/jvm/R tests and run them manually when we are confident.  As passing the older version usually means it can pass in newer version.
- Move CPU tests to some free services.

Reply:
All good suggestions. I would like to put in some effort to trim down some CI tests. Stay tuned.